example/src/main/scala/example/Example.scala

```diff
@@ -0,0 +1,66 @@
+package example
+
+import cats.Monad
+import cats.syntax.flatMap._
+import slog.{LoggerFactory, LoggingContext}
+
+class Example[F[_]](loggerFactory: LoggerFactory[F],
+                    loggingContext: LoggingContext[F])(implicit F: Monad[F]) {
+  private[this] val logger = loggerFactory.make[Example[F]]
+  // or
+  // private [this] val logger = loggerFactory.make("my-logger")
+
+  // example output json does not include timestamps, log levels, thread name,
+  // logger names etc to keep things simple. They are of course included in the real
+  // output.
+  def foo: F[Unit] = {
+    // simple log message. Might take from scope.
+    // {"message": "Hello world!!", "file": "Example.scala", "line": 19}
+    logger.info.msg("Hello world!!") >>
+      // Provide extract context to the log message
+      // {"message": "Hello", "file": "Example.scala", "line": 22, "correlation_id": "<VALUE>"}
+      logger.info.context("correlation_id", "<VALUE>").msg("Hello") >>
```

Comment:
i wonder if this would be a good usecase for `HList` ... calling `context` all the time adds a lot of visual noise - having `HList[(String, SomethingSerializable)]` might be the way to go

Reply:
You still need to provide keys. And shapeless is not super user friendly.

Reply:
I also think we should rather have fluent interface than HList because of use-friendliness